pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -169,20 +173,28 @@ func buildOutboundNetworkFiltersWithWeightedClusters(env *model.Environment, nod
 		}
 	}
 
+	// TODO: Need to handle multiple cluster names for Redis
+	clusterName := clusterSpecifier.WeightedClusters.Clusters[0].Name
 	tcpFilter := setAccessLogAndBuildTCPFilter(env, node, proxyConfig)
-	return buildNetworkFiltersStack(node, port, tcpFilter, statPrefix)
+	return buildNetworkFiltersStack(node, port, tcpFilter, statPrefix, clusterName)
 }
 
 // buildNetworkFiltersStack builds a slice of network filters based on
 // the protocol in use and the given TCP filter instance.
-func buildNetworkFiltersStack(node *model.Proxy, port *model.Port, tcpFilter *listener.Filter, statPrefix string) []listener.Filter {
+func buildNetworkFiltersStack(node *model.Proxy, port *model.Port, tcpFilter *listener.Filter, statPrefix string, clusterName string) []listener.Filter {
 	filterstack := make([]listener.Filter, 0)
 	switch port.Protocol {
 	case model.ProtocolMongo:
-		filterstack = append(filterstack, buildMongoFilter(statPrefix, util.IsProxyVersionGE11(node)), *tcpFilter)
+		filterstack = append(filterstack, buildMongoFilter(statPrefix, util.IsXDSMarshalingToAnyEnabled(node)), *tcpFilter)
+	case model.ProtocolRedis:
+		if util.IsProxyVersionGE11(node) && pilot.EnableRedisFilter() {
+			filterstack = append(filterstack, buildRedisFilter(statPrefix, clusterName, util.IsXDSMarshalingToAnyEnabled(node)))
+		} else {
+			filterstack = append(filterstack, *tcpFilter)
```

Comment:
I realize you just undid https://github.com/istio/istio/pull/12236 here basically

But I am wondering why for redis filter we apply either Redis or TCP, but for mongo and mysql we add both?

Assuming that is intentional, lgtm, thanks for re-enabling this

Reply:
Good catch, i think for redis, it should be same,otherwise,  there will be no access logs